engine/src/main/java/org/terasology/rendering/nui/widgets/UIText.java

```diff
@@ -667,7 +674,12 @@ public void bindText(Binding<String> binding) {
      * @return The text contained by the text box
      */
     public String getText() {
-        return text.get();
+        String arrayText = "";
+        if(textArr != null){
+            for(int i = 0;i < textArr.length;i++) arrayText += textArr[i] + "\n";
```

Comment:
I'd replace this with a String join call -- it's more robust and is easier to read.